shared/js/background/chrome-events.es6.js

```diff
@@ -207,42 +221,61 @@ chrome.runtime.onMessage.addListener((req, sender, res) => {
  */
 const agents = require('./storage/agents.es6')
 const agentSpoofer = require('./classes/agentspoofer.es6')
+// TODO fix for manifest v3
+const sessionKey = getHash()
 
-// Inject fingerprint protection into sites when
-// they are not whitelisted.
-chrome.webNavigation.onCommitted.addListener(details => {
-    let tab = tabManager.get({ tabId: details.tabId })
-    if (tab && tab.site.isBroken) {
-        console.log('temporarily skip fingerprint protection for site: ' + details.url +
-          'more info: https://github.com/duckduckgo/content-blocking-whitelist')
-        return
-    }
-    if (tab && !tab.site.whitelisted) {
-        // Set variables, which are used in the fingerprint-protection script.
-        try {
-            const variableScript = {
-                'code': `
-                    try {
-                        var ddg_ext_ua='${JSON.stringify(agentSpoofer.getAgent())}'
-                        var ddg_referrer=${JSON.stringify(tab.referrer)}
-                    } catch(e) {}`,
-                'runAt': 'document_start',
-                'frameId': details.frameId,
-                'matchAboutBlank': true
-            }
-            chrome.tabs.executeScript(details.tabId, variableScript)
-            const scriptDetails = {
-                'file': '/data/fingerprint-protection.js',
-                'runAt': 'document_start',
-                'frameId': details.frameId,
-                'matchAboutBlank': true
+async function getContentScope () {
+    const url = chrome.runtime.getURL('/public/js/content-scope.js')
+
+    let response = await fetch(url)
+    return response.text()
+}
+
+async function init () {
+    const contentScopeScript = await getContentScope()
```

Comment:
@Charlie-belmer so I moved the code to be only called here once which shouldn't slow the performance of anything, the only tiny impact will be setup of the background script listener below will be delayed on install / update of the extension.

The end goal being all code in fingerprint-protection.js will move into it's own file in: shared/js/content-scope allowing us to lint it properly.
This move eventually would give us the ability to inject fingerprint-protection.js always statically; but activate it in this listener, which should reduce the amount of chance of a race condition.